### PR TITLE
Feature/visibility

### PR DESCRIPTION
**Pull request by FordMcLeod**
Implemented visibility proper, still need to monitor the private url sharing things as the frontend input method changes and the user URL as ID's are modified. As noted in helpers.py.

### CODE REVIEW

File: posts/helpers.py [hunk 1]
```diff
@@ -18,6 +20,14 @@ def parse_id_from_url(url):
     path = path.split('/')
     return path[-1]
 
+def get_url(user):
+    # TODO: When the serializer is unfucked regarding domain this should use the serializer to spit out the url :)
+    # YES this shouldn't be localhost but let's talk when the urls for authors stop being strange 
+    userSerializer = UserSerializer(user)
+    #url = userSerializer.get_absolute_url()
+    url = "http://127.0.0.1:8000/author/" + str(user.id)
```
Comment (MichaelParadis) [inline posts/helpers.py]: You should be able to use this. The example.com is due to your configuration.

File: posts/helpers.py [hunk 2]
```diff
@@ -45,16 +55,53 @@ def are_FOAF(user, other):
...
+        return False
+
+def is_local_user(url):
+    userHost = urlparse(url).hostname
+    return (userHost == Site.objects.get_current().domain)
+
+def visible_to(post,user, direct=False):
+    author = get_user(post.author.id)
+    owns_post = (user == author)
+    if (owns_post):
```
Comment (MichaelParadis) [inline posts/helpers.py]: Couldn't this just be user == author?

File: posts/urls.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
 from .views import PostCreateView
 from .viewsfolder.feed_views import FrontEndPublicPosts
 from .viewsfolder.feed_views import FrontEndAuthorPosts
+from .viewsfolder.feed_views import FrontEndFeed
```
Comment (MichaelParadis) [inline posts/urls.py:18]: https://docs.djangoproject.com/en/2.1/ref/settings/#append-slash
Add this to the setting file please.

**Review by MichaelParadis — CHANGES_REQUESTED** [review]

File: posts/urls.py (hunk 3 above)
Comment (FordMcLeod) [reply posts/urls.py:18]: if you just mean putting in "APPEND_SLASH = True" into settings.py, it is done :)

Comment (MichaelParadis) [reply posts/urls.py]:
```suggestion
from .viewsfolder.feed_views import FrontEndFeedThat`
````
Sweet.

Comment (MichaelParadis) [reply posts/urls.py:18]: Sweet

**Comment by MichaelParadis** [comment]
@FordMcLeod can you fix the conflicts?

Will close #63

**Comment by FordMcLeod** [comment]
Yeah yeah I can, I'm just fixing the url bidness now.

**Comment by FordMcLeod** [comment]
Let me rephrase, I already fixed conflicts, just fixing a small url thing
almost done

**Comment by FordMcLeod** [comment]
Done. just fixed the url BS i was having

**Review by MichaelParadis — APPROVED** [review]
